### PR TITLE
assert: fix build break

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

assert: fix build break

misc/assert.c: In function 'show_tasks':
Error: misc/assert.c:411:10: error: format '%p' expects argument of type 'void *', but argument 3 has type 'uintptr_t' {aka 'long unsigned int'} [-Werror=format=]
  411 |   _alert("  ----   ---"
      |          ^~~~~~~~~~~~~~
......
  424 |          , up_get_intstackbase()
      |            ~~~~~~~~~~~~~~~~~~~~~
      |            |
      |            uintptr_t {aka long unsigned int}

Signed-off-by: ligd <liguiding1@xiaomi.com>

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — COMMENTED** [review]
What about 64bit architectures? Or we expect pointers only in lower 32 bit address space?

**Comment by pkarashchenko** [comment]
Why not just to cast to `void *` and leave `%p`?

**Comment by xiaoxiang781216** [comment]
> What about 64bit architectures? Or we expect pointers only in lower 32 bit address space?

uintptr_t can handle both 32bit and 64bit arch


> Why not just cast to `void *` and leave `%p`?

%p can't add the modifier (e.g., 08).

**Review by jerpelea — APPROVED** [review]

**Comment by GUIDINGLI** [comment]
@xiaoxiang781216 @pkarashchenko @jerpelea 
Please help to re-trigger, this is download fail

**Comment by xiaoxiang781216** [comment]
> > > What about 64bit architectures? Or we expect pointers only in lower 32 bit address space?
> > 
> > 
> > uintptr_t can handle both 32bit and 64bit arch
> > > Why not just cast to `void *` and leave `%p`?
> > 
> > 
> > %p can't add the modifier (e.g., 08).
> 
> But shouldn't it be 16 modifier for 64 bit pointers?

Oh, yes you are right.

File: sched/misc/assert.c [hunk 1]
```diff
@@ -329,7 +331,7 @@ static void dump_task(FAR struct tcb_s *tcb, FAR void *arg)
 #ifdef CONFIG_SMP
          "  %4d"
 #endif
-         "   %p"
+         "   %0*p"
```
Comment (pkarashchenko) [inline sched/misc/assert.c:334]: Build complaints somewhere around here

**Pull request merged**